main.rb

```diff
@@ -0,0 +1,149 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength, Style/For
+module Enumerable
+  # 1.my_each
+  def my_each
+    return to_enum unless block_given?
+
+    for i in self
+      yield(i)
+    end
+    self
+  end
+
+  # 2. my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    for i in (0..self).size - 1
+      yield(self[i], i)
+    end
+    self
+  end
```

Comment:
## my_each_with_index:

- [ ] You have an error in your my_each_with_index method, please fix it:

![image](https://user-images.githubusercontent.com/34653764/101418525-61da4b00-38ee-11eb-8915-aac71c0b57e8.png)